### PR TITLE
ILM use Priority.IMMEDIATE for stop ILM cluster update

### PR DESCRIPTION
**Pull request by andreidan**
This changes the priority of the cluster state update that stops ILM
altogether to `IMMEDIATE`. We've chosen to change this as it can be useful to
temporarily stop ILM if a cluster is overwhelmed, but a `NORMAL`
priority can see the "stop ilm update" not make it up the tasks queue.

On the same note, we're keeping the `start ILM` cluster update priority
to `NORMAL` on purpose such that we only start `ILM` if the cluster can
handle it.

### CODE REVIEW

**Comment by andreidan** [comment]
@elasticmachine update branch

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportStopILMAction.java [hunk 1]
```diff
@@ -50,7 +51,8 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, StopILMRequest request, ClusterState state, ActionListener<AcknowledgedResponse> listener) {
         clusterService.submitStateUpdateTask("ilm_operation_mode_update",
-                new AckedClusterStateUpdateTask<AcknowledgedResponse>(request, listener) {
+                new AckedClusterStateUpdateTask<AcknowledgedResponse>(Priority.URGENT, request, listener) {
```
Comment (dakrone) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportStopILMAction.java:54]: This also needs to change the cluster state update on line 366 of IndexLifecycleService (`submitOperationModeUpdate`), that is the second half that moves the state from `STOPPING` -> `STOPPED`

**Review by dakrone — CHANGES_REQUESTED** [review]
Thanks @andreidan, this needs to update the second half of the shutdown to also use the priority though, so both parts are urgent

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportStopILMAction.java (hunk 1 above)
Comment (andreidan) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportStopILMAction.java:54]: Ah yes, good shout

**Comment by DaveCTurner** [comment]
I suggest `IMMEDIATE` rather than `URGENT` for this task. I don't know of any starvation issues at `URGENT` priority but my concern is that we may uncover such an issue in future which prevents us from stopping ILM in a timely fashion. This action is very lightweight, only available to admins, rarely called, but useful when the cluster is under severe pressure, and I can't think of a reason for spamming it that isn't obviously abusive.

Please sprinkle some comments throughout these cluster state updates indicating that they may be running at `IMMEDIATE` priority, as a defence against accidentally making them more heavyweight in future.

**Comment by DaveCTurner** [comment]
Also FWIW you can test that the task really runs at a high enough priority with something like this:

https://github.com/elastic/elasticsearch/blob/95a7eed9aa35f47b228e402508709b5bd6703cf4/server/src/test/java/org/elasticsearch/cluster/ClusterHealthIT.java#L268-L287

(replace `Priority.LOW` with `Priority.IMMEDIATE`) -- even if the cluster is processing this looping task it should still process other `IMMEDIATE` tasks but nothing lower. I prefer that to simply asserting that it returns the right `.priority()`.

**Comment by andreidan** [comment]
@DaveCTurner thanks for pointing out the `ClusterHealthIT`. I wasn't aware of it.

If you don't have a strong opinion on adding an integration test, I'd rather not, as ILM is a client of the `ClusterService`, using the `ClusterService` API according to the specification (I think ensuring the order and fairness of task execution is something that we should, and probably are, test as part of the `cluster` module - I _think_ the `PrioritizedExecutorsTests` is making sure the executor invariants hold)

**Review by dakrone — APPROVED** [review]
LGTM, thanks Andrei

**Pull request merged**